Podcast/SearchViewController.swift

```diff
@@ -240,21 +241,36 @@ class SearchViewController: ViewController, UISearchControllerDelegate, UITableV
             }
             return
         }
-        if searchText == lastSearchText { return }
-        lastSearchText = searchText
-        tabUnderlineView.isHidden = false
-        searchResultsTableView.isHidden = false
-        pastSearchesTableView.isHidden = true
-        didDismissItunesHeaderForQuery = false
-        addSearchITunesHeader()
-        mainScrollView = searchResultsTableView
-        searchResultsTableView.startLoadingAnimation()
-        for data in tableViewData {
-            data.fetchData(query: searchText, newSearch: true)
+        if self.lastSearchText == searchText && pastSearchesTableView.isHidden { return }
+        self.lastSearchText = searchText
+        self.tabUnderlineView.isHidden = false
+        self.searchResultsTableView.isHidden = false
+        self.pastSearchesTableView.isHidden = true
+        self.didDismissItunesHeaderForQuery = false
+        self.addSearchITunesHeader()
+        self.mainScrollView = self.searchResultsTableView
+        self.searchResultsTableView.startLoadingAnimation()
```

Comment:
why do these all use `self`?

Reply:
thanks for catching that! I had it in a completion block before